### PR TITLE
Resolved #171

### LINKED ISSUE #171: Add Pro Micro pin names image to documentation
We often get questions on Discord about the naming for the `pro_micro_d` and `pro_micro_a` [nexus nodes](https://docs.zephyrproject.org/2.3.0/guides/porting/shields.html#gpio-nexus-nodes) we have in use for ZMK.

We should update the docs to include the attached CC licensed image 
pro-micro-pins

On the following pages:

* https://zmkfirmware.dev/docs/dev-boards-shields-keymaps#pro-micro-compatible-keyboard
* https://zmkfirmware.dev/docs/dev-guide-new-shield#shield-overlay

### PR DESCRIPTION
**Pull request by LowNightSnack**
Thank you @petejohanson for this opportunity. I resolved issue #171.

### CODE REVIEW

File: docs/docs/dev-guide-new-shield.md [hunk 1]
```diff
@@ -64,6 +64,8 @@ endif
 
 ## Shield Overlay
 
+![Labelled Pro Micro pins](pro-micro-labelled-image.jpg)
```
Comment (petejohanson) [inline docs/docs/dev-guide-new-shield.md:67]: In addition to the image itself, we should include some description of how the DT nodes map to the labelled pins in the image., Perhaps something along the lines of:

"ZMK uses the green Arduino pin names ,e.g. `D0`, or `A1`, as the reference to generate references to specific pro-micro pin locations. For example, the location labeled `D0` in this diagram can be referred to in the ZMK devicetree files as `&pro_micro_d 0`."

Thoughts?

**Review by petejohanson — CHANGES_REQUESTED** [review]
Thanks for the quick PR! One request for changes. Thanks!

File: docs/docs/dev-guide-new-shield.md (hunk 1 above)
Comment (LowNightSnack) [reply docs/docs/dev-guide-new-shield.md:67]: I could add that to the documentation. I will write some improvements and post it here. Though now i have a fair idea of what is happening thanks to you.

Comment (LowNightSnack) [reply docs/docs/dev-guide-new-shield.md:67]: "ZMK uses the green color coded pin names to generate devicetree node references. For example, to refer to the node `D0` in the devicetree files, use `&pro_micro_d 0`."
Is this okay?

Comment (petejohanson) [reply docs/docs/dev-guide-new-shield.md]: Can we give a second example for an A pin as well? Want to be sure we cover that as well. Thanks!

Comment (LowNightSnack) [reply docs/docs/dev-guide-new-shield.md]: "ZMK uses the green color coded pin names to generate devicetree node references. For example, to refer to the node `D0` in the devicetree files, use `&pro_micro_d 0` or to refer to `A1`, use `&pro_micro_a 1`."
Does this work?

File: docs/docs/behavior/hold-tap.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ A code example which configures a mod-tap setting that works with homerow mods:
 
 If this config does not work for you, try the flavor "tap-preferred" and a short tapping_term_ms such as 120ms.
 
-If you want to use a tap-hold with a keycode from a different code page, you have to define another behavior with another "bindings" parameter.For example, if you want to use SHIFT and volume up, define the bindings like `bindings = <&kp>, <&cp>;`. Only single-argument behaviors are supported at the moment.
```
Comment (petejohanson) [inline docs/docs/behavior/hold-tap.md:59]: Can we take out these unrelated changed that snuck in here and a few other files please? Thanks.

File: docs/docs/dev-clean-room.md [hunk 3]
```diff
@@ -11,7 +11,7 @@ Anyone wanting to contribute code to ZMK _must_ read this, and adhere to the ste
 :::
 
 ZMK Firmware is a [clean room design](https://en.wikipedia.org/wiki/Clean_room_design) keyboard firmware, that
-borrows/implements a lot of the features found in popular keyboard firmwares projects like [QMK](https://qmk.fm)
```
Comment (petejohanson) [inline docs/docs/dev-clean-room.md:14]: Ditto.

File: docs/docs/user-setup.md [hunk 4]
```diff
@@ -179,7 +179,7 @@ storage device. Once the flash is complete, the controller should automatically
 
 ## Wirelessly Connecting Your Keyboard
 
-Connecting your keyboard wirelessly is the same as adding other Bluetooth devides: press the reset button and scan for devices. However, pairing and bonding is still currently being worked on to increase relability and ease of use. In addition, users have in general reported that Bluetooth pairing with computers tends to be quite finnicky. Try out the connection with your tablet or phone first, as those devices seem to work much more consistently. See [BLE Reset](./bond-reset.md) for help on resetting your MCUs if you're experiencing connection issues.
```
Comment (petejohanson) [inline docs/docs/user-setup.md:182]: And here.

**Review by petejohanson — CHANGES_REQUESTED** [review]
A few unrelated change snuck in. Can you please revert?

**Comment by LowNightSnack** [comment]
All the I changes made were reverted and the requested block was added to the dev-guide-new-shield.md
Please check if we are good to go.
I am sorry for the inconvenience caused by me.

File: docs/docs/behavior/hold-tap.md [hunk 5]
```diff
@@ -59,4 +59,4 @@ If this config does not work for you, try the flavor "tap-preferred" and a short
 If you want to use a tap-hold with a keycode from a different code page, you have to define another behavior with another "bindings" parameter.For example, if you want to use SHIFT and volume up, define the bindings like `bindings = <&kp>, <&cp>;`. Only single-argument behaviors are supported at the moment.
 
 #### Note
-Astute readers may notice similarities between the possible behaviors in ZMK and other firmware, such as QMK. The hold-preferred flavor works similar to the `HOLD_ON_OTHER_KEY_PRESS` setting. The 'balanced' flavor is similar to the `PERMISSIVE_HOLD` setting, and the `tap-preferred` flavor is similar to `IGNORE_MOD_TAP_INTERRUPT`.
```
Comment (petejohanson) [inline docs/docs/behavior/hold-tap.md:62]: Just one more lingering change that snuck in, then we're good to go. Thanks!

Comment (LowNightSnack) [reply docs/docs/behavior/hold-tap.md:62]: In the rich diff view, there seems to be no changes to display. Could you confirm that?

Comment (petejohanson) [reply docs/docs/behavior/hold-tap.md:62]: My suspicion is maybe it's just a line ending change?

You can try something along the lines of (assuming you have the ZMK repo as `origin` remote):

```
git checkout origin/main docs/docs/behavior/hold-tap.md
```

(For reference: https://stackoverflow.com/a/1817774)

Comment (LowNightSnack) [reply docs/docs/behavior/hold-tap.md:62]: I am not doing on my remote machine. And i have tried every other thing except for deleting the file and re-uploading.
Man, this does makes my commit history look messed up. Sorry about that.
I will keep this in mind next time.

Comment (petejohanson) [reply docs/docs/behavior/hold-tap.md:62]: No worries. Sorry you're having such trouble. No reason to hold up getting the additional docs in for this. Thanks!

**Review by petejohanson — APPROVED** [review]
Thanks!

**Pull request merged**